R/ExtractData_Soils.R

```diff
@@ -263,6 +263,212 @@ extract_soil_CONUSSOIL <- function(MMC, sim_size, sim_space, dir_ex_soil,
   MMC
 }
 
+#' Extracts all .tif files' data (clay, sand, bd, gravel, depth, ...) in
+#' dir_ex_soil to an environment
+#' @param MMC An environment containing the structure for extracted data to be
+#'   extracted to
+#' @param sim_size An environment containing information on the runs.
+#' @param sim_space An environment containing information on site locations and
+#'   extraction type, ie. cell or point.
+#' @param dir_ex_soil String for the location of the soil files to be
+#'  extracted.
+#' @param fnames_in An environment containing output file locations.
+#' @param resume Logical whether or not to resume the project if it failed
+#'  partially through before.
+#' @param verbose Logical whether to see additional messages or not as the
+#'  function executes.
+#' @param default_TOC_GperKG A numeric value. The default value is
+#'   0 g \var{TOC} per kg soil.
+#' @return MMC An environment containing the extracted data.
+#' @references Hengl. T. et al, 2017. SoilGrids250m: Global gridded soil
+#'  information based on machine learning.
+#' @references Shangguan, W. et al, 2016. Mapping the global depth to bedrock
+#' for land surface modeling.
+#' @author Nathan Payton- McCauslin. July 2018.
+#' @section Note: \enumerate{
+#' \item All data is taken at a resolution of 250m.
+#'     \item Data Extracted Using Reference 1: \itemize{
+#'     \item Sand
+#'     \item Clay
+#'     \item Bulk Density
+#'     \item Gravel
+#'    }
+#'  \item Data Extracted Using Reference 2: \itemize{
+#'      \item Depth to Bedrock
+#'    }
+#'  }
+extract_soil_ISRIC250m <- function(MMC, sim_size, sim_space,
+                                       dir_ex_soil, fnames_in, resume, verbose,
+                                       default_TOC_GperKG = 0){
+
+  # print stats
+  if (verbose) {
+    t1 <- Sys.time()
+    temp_call <- shQuote(match.call()[1])
+    print(paste0("rSFSW2's ", temp_call, ": started at ", t1))
+    on.exit({
+            print(paste0("rSFSW2's ", temp_call, ": ended after ",
+                          round(difftime(Sys.time(), t1, units = "secs"), 2),
+                          " s")) ; cat("\n")}, add = TRUE)
+  }
+  # set up data files for extraction =============================
+  show_site_info <- verbose
+  soil_layer <- 1
+  dir.ex.gridded <- file.path(dir_ex_soil, "Isric", "GriddedGlobalV5")
+  # stop program execution if folder path is incorrect
+  if (!file.exists(dir.ex.gridded)) stop(paste0("Folder '", dir.ex.gridded,
```

Comment:
Use `dir.exists` instead of `file.exists` to check the existence of a directory -- they should behave the same on POSIX-compliant systems, but this is not guaranteed on Windows-type systems; see `?file.exists`